index.js

```diff
@@ -274,9 +274,13 @@ module.exports = class Hyperswarm extends EventEmitter {
     const existing = this._allConnections.get(conn.remotePublicKey)
 
     if (existing) {
+      // If both connections are from the same peer,
+      // - pick the new one if the existing stream is already established (has sent and received bytes),
+      //   because the other client must have lost that connection and be reconnecting
+      // - otherwise, pick the one thats expected to initiate in a tie break
+      const existingIsOutdated = existing.rawBytesRead > 0 && existing.rawBytesWritten > 0
```

Comment:
@mafintosh Currently, `hyperswarm` doesn't depend on `@hyperswarm/secret-stream` directly. However, this code depends on features introduced in secret-stream 6.6.0. How do we generally deal with such in this org? 

1. I can make it backwards compatible, e.g.
```suggestion
      const existingIsOutdated = typeof existing.rawBytesRead === 'number' && (existing.rawBytesRead > 0 && existing.rawBytesWritten > 0)
```

2. We can add @hyperswarm/secret-stream > 6.6.0 as a peer-dep
3. or as a regular dep
4. or bump the version numbers of hyperdht even though no hyperdht code has changed
5. or say screw it...

Reply:
waiting on a publish of hyperdht 6.15.3 then

Reply:
was told on slack to yolo it, so I'm going with option 5